lib/rubocop/cop/rspec/factory_bot/attribute_defined_statically.rb

```diff
@@ -27,47 +27,12 @@ module FactoryBot
         class AttributeDefinedStatically < Cop
           MSG = 'Use a block to declare attribute values.'.freeze
 
-          ATTRIBUTE_DEFINING_METHODS = %i[factory trait transient ignore].freeze
-
-          UNPROXIED_METHODS = %i[
-            __send__
-            __id__
-            nil?
-            send
-            object_id
-            extend
-            instance_eval
-            initialize
-            block_given?
-            raise
-            caller
-            method
-          ].freeze
-
-          DEFINITION_PROXY_METHODS = %i[
-            add_attribute
-            after
-            association
-            before
-            callback
-            ignore
-            initialize_with
-            sequence
-            skip_create
-            to_create
-          ].freeze
-
-          RESERVED_METHODS =
-            DEFINITION_PROXY_METHODS +
-            UNPROXIED_METHODS +
-            ATTRIBUTE_DEFINING_METHODS
-
           def_node_matcher :value_matcher, <<-PATTERN
-            (send {self nil?} !#reserved_method? $...)
+            (send { self nil? (lvar #capture_arg) } !#reserved_method? $...)
```

Comment:
Generally, you can use `%` in the pattern to pass variables, and avoid using the instance variable this way

Reply:
Interesting. I am unsure how to do use that. Is there documentation somewhere? Or an example?

I realized that when I took the suggestion to use instance variables I also introduced a regression that hadn't been tested, so I added a test case for it and went back to the version I had before for now.